### PR TITLE
Added created_at and updated_at fields into trigger

### PR DESCRIPTION
**Pull request by kissken**
added new fields into trigger: 

- created_at (null if not set)
- updated_at (null if not set)

### CODE REVIEW

File: database/redis/trigger.go [hunk 1]
```diff
@@ -114,17 +114,15 @@ func (connector *DbConnector) RemovePatternTriggerIDs(pattern string) error {
 // If given trigger contains new tags then create it.
 // If given trigger has no subscription on it, add it to triggers-without-subscriptions
 func (connector *DbConnector) SaveTrigger(triggerID string, trigger *moira.Trigger) error {
-	if trigger.IsRemote {
```
Comment (kissken) [inline database/redis/trigger.go:117]: It was moved into preSave func.

File: database/redis/trigger_test.go [hunk 2]
```diff
@@ -688,6 +710,53 @@ func TestTriggerErrorConnection(t *testing.T) {
...
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	systemClock := mock_clock.NewMockClock(mockCtrl)
+	systemClock.EXPECT().Now().Return(testTime).Times(4)
+	connector := &DbConnector{clock: systemClock}
+	patterns := []string{"pattern-1", "pattern-2"}
+
+	Convey("Test preSave function for local trigger", t, func() {
+		trigger := &moira.Trigger{ID: "trigger-id", Patterns: patterns}
+		Convey("Test preSave function for creation local trigger", func() {
```
Comment (zhelyabuzhsky) [inline database/redis/trigger_test.go:725]:
```suggestion

		Convey("Test preSave function for creation local trigger", func() {
```

File: interfaces.go [hunk 3]
```diff
@@ -217,3 +217,8 @@ type PlotTheme interface {
 	GetXAxisStyle() chart.Style
 	GetYAxisStyle() chart.Style
 }
+
+// Clock is an interface to get time.Now()
```
Comment (zhelyabuzhsky) [inline interfaces.go:221]: Nit: Lets add more universal comment pls)

File: database/redis/trigger_test.go (hunk 2 above)
Comment (zhelyabuzhsky) [reply database/redis/trigger_test.go:725]: Check the official library example pls )

https://github.com/smartystreets/goconvey/blob/master/examples/bowling_game_test.go

File: api/dto/triggers.go [hunk 4]
```diff
@@ -73,6 +73,10 @@ type TriggerModel struct {
 	MuteNewMetrics bool `json:"mute_new_metrics"`
 	// A list of targets that have only alone metrics
 	AloneMetrics map[string]bool `json:"alone_metrics"`
+	// Datetime of  creation
```
[the remark this answers is not in the document]
Comment (zhelyabuzhsky) [reply api/dto/triggers.go:76]: Nit )

Or just remove extra space

File: api/dto/triggers.go [hunk 5]
```diff
@@ -73,6 +73,10 @@ type TriggerModel struct {
...
+	CreatedAt *time.Time `json:"created_at"`
+	// Update datetime
```
[the remark this answers is not in the document]
Comment (zhelyabuzhsky) [reply api/dto/triggers.go:78]: Nit

File: api/dto/triggers.go [hunk 6]
```diff
@@ -94,6 +98,8 @@ func (model *TriggerModel) ToMoiraTrigger() *moira.Trigger {
 		IsRemote:       model.IsRemote,
 		MuteNewMetrics: model.MuteNewMetrics,
 		AloneMetrics:   model.AloneMetrics,
+		CreatedAt:      getUnixTime(model.CreatedAt),
```
Comment (zhelyabuzhsky) [inline api/dto/triggers.go:101]: ?

File: database/redis/trigger.go [hunk 7]
```diff
@@ -195,6 +193,19 @@ func (connector *DbConnector) updateTrigger(triggerID string, newTrigger *moira.
...
+	if newTrigger.IsRemote {
+		newTrigger.Patterns = make([]string, 0)
+	}
+
+	now := time.Now().Round(0)
+	newTrigger.UpdatedAt = &now
+	newTrigger.CreatedAt = &now
+	if oldTrigger != nil {
+		newTrigger.CreatedAt = oldTrigger.CreatedAt
+	}
```
[the remark this answers is not in the document]
Comment (zhelyabuzhsky) [reply database/redis/trigger.go:206]: nit

**Review by zhelyabuzhsky — CHANGES_REQUESTED** [review]
molorik, tol'ko...

File: api/dto/triggers.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (kissken) [reply api/dto/triggers.go]: changed comment

File: database/redis/trigger_test.go (hunk 2 above)
Comment (kissken) [reply database/redis/trigger_test.go]: rewrote

File: interfaces.go (hunk 3 above)
Comment (kissken) [reply interfaces.go]: changed

File: api/dto/triggers.go (hunk 6 above)
Comment (kissken) [reply api/dto/triggers.go]: deleted

File: database/redis/trigger_test.go [hunk 8]
```diff
@@ -47,7 +51,7 @@ func TestTriggerStoring(t *testing.T) {
 			So(hasSubscriptions, ShouldBeFalse)
 		})
 
-		Convey("Test save-get-remove", func() {
+		FocusConvey("Test save-get-remove", func() {
```
Comment (zhelyabuzhsky) [inline database/redis/trigger_test.go:54]: oops

**Review by dmitryanchikov — APPROVED** [review]
LGTM!

**Pull request merged**